liquibase-core/src/main/java/liquibase/Liquibase.java

```diff
@@ -643,30 +403,44 @@ protected RollbackVisitor createRollbackVisitor(List<ChangesetsRolledback.Change
         return new RollbackVisitor(database, changeExecListener, processedChangesets);
     }
 
-    protected ChangeLogIterator getStandardChangelogIterator(Contexts contexts, LabelExpression labelExpression,
-                                                             DatabaseChangeLog changeLog) throws DatabaseException {
-       return getStandardChangelogIterator(contexts, labelExpression, false, changeLog);
-    }
-
     /**
      *
      * Return a ChangeLogIterator constructed with standard filters
      *
      * @param   contexts                           Contexts to filter for
      * @param   labelExpression                    Labels to filter for
-     * @param   collectAllReasons                  Flag to control whether all skip reasons are accumulated
-     *                                             default value is false to only gather the first
      * @param   changeLog                          The changelog to process
      *
      * @return  ChangeLogIterator
      * @throws DatabaseException
      *
      */
     protected ChangeLogIterator getStandardChangelogIterator(Contexts contexts, LabelExpression labelExpression,
-                                                             boolean collectAllReasons,
                                                              DatabaseChangeLog changeLog) throws DatabaseException {
         return new ChangeLogIterator(changeLog,
-                collectAllReasons,
+                new ShouldRunChangeSetFilter(database),
+                new ContextChangeSetFilter(contexts),
+                new LabelChangeSetFilter(labelExpression),
+                new DbmsChangeSetFilter(database),
+                new IgnoreChangeSetFilter());
+    }
+
+    /**
+     *
+     * Return a StatusChangeLogIterator constructed with standard filters
+     *
+     * @param   contexts                           Contexts to filter for
+     * @param   labelExpression                    Labels to filter for
+     *                                             default value is false to only gather the first
+     * @param   changeLog                          The changelog to process
+     *
+     * @return  StatusChangeLogIterator
+     * @throws DatabaseException
+     *
+     */
+    protected ChangeLogIterator getStandardStatusChangelogIterator(Contexts contexts, LabelExpression labelExpression,
+                                                                   DatabaseChangeLog changeLog) throws DatabaseException {
+        return new StatusChangeLogIterator(changeLog,
```

Comment:
I don't think these methods that you added are used anywhere

Reply:
Removed getStandardStatusChangelogIterator